### PR TITLE
gRPC instrumentation: client additions

### LINKED ISSUE #256: gRPC client instrumentation example doesn't run
The example in the gRPC instrumentation for the client side doesn't actually run as is.

**Steps to reproduce**
I copied the code into a Python file, made up some generated protobuf code for testing, and ran the program.

**What is the expected behavior?**
```
Greeter client received: YOU
```

**What is the actual behavior?**
```Traceback (most recent call last):
  File "/tmp/client.py", line 42, in <module>
    run()
  File "/tmp/client.py", line 32, in run
    with grpc.insecure_channel("localhost:50051") as channel:
  File "/home/michael/work/opentelemetry-python-contrib/instrumentation/opentelemetry-instrumentation-grpc/src/opentelemetry/instrumentation/grpc/grpcext/_interceptor.py", line 235, in __enter__
    raise NotImplementedError()
NotImplementedError
```

I'll probably work on this one too, since I need a test client to verify some of the other issues I'm already working on.

### PR DESCRIPTION
**Pull request by alertedsnake**
# Description

Updates the gRPC client instrumentation:

* Add additional attributes to span described in the spec but not previously added
* fix example in docs that didn't work #256
* rework the instrumentation call so both secure and insecure connections are instrumented by default - otherwise there's no way to handle both
* properly capture and record RPC errors
* change all metrics labels recorded to be strings
* change metrics labels to be namespaced like the span attributes are, this seems to be a convention when looking at other client instrumentation (i.e. requests)

Re: the instrumentation call - I preserved the existing parameter so as to not break changes, but we may wish to mark that as deprecated, I don't know the convention with this library on how to do that.

Fixes #256 

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

I've added new checks in the existing tests to make sure the attributes 

# Does This PR Require a Core Repo Change?

- [X] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/master/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [X] Followed the style guidelines of this project
- [X] Changelogs have been updated
- [X] Unit tests have been added
- [X] Documentation has been updated

### CODE REVIEW

**Comment by alertedsnake** [comment]
Is there anything you need me to do to help the process along here?  This gRPC part of this is not really usable as-is, and I'd like to help with other things, but I'm blocked by these PRs.

**Review by codeboten — APPROVED** [review]
Code changes LGTM

**Comment by alertedsnake** [comment]
The docker error in the tests seems entirely unrelated to the changes made in this PR, does anyone know how to fix?

**Comment by ocelotl** [comment]
> The docker error in the tests seems entirely unrelated to the changes made in this PR, does anyone know how to fix?

I am investigating it :mag_right:

**Comment by alertedsnake** [comment]
> I am investigating it

Thanks!

**Review by ocelotl — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-grpc/src/opentelemetry/instrumentation/grpc/__init__.py [hunk 1]
```diff
@@ -185,29 +185,33 @@ class GrpcInstrumentorClient(BaseInstrumentor):
 
     """
 
+    # Figures out which channel type we need to wrap
+    def _which_channel(self, kwargs):
+        # handle legacy argument
+        if "channel_type" in kwargs:
+            if kwargs.get("channel_type") == "secure":
+                return ("secure_channel",)
```
[the remark this answers is not in the document]
Comment (alertedsnake) [reply instrumentation/opentelemetry-instrumentation-grpc/src/opentelemetry/instrumentation/grpc/__init__.py:193]: Oh! I didn't even realize.  I'll fix it, good catch.

**Pull request merged**